seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java

```diff
@@ -34,16 +34,18 @@ public final class SeaTunnelRow implements Serializable {
     /** The array to store the actual internal format values. */
     private final Object[] fields;
 
-    private volatile int size;
-
     private Map<String, Object> options;
 
+    private volatile int size;
+
     public SeaTunnelRow(int arity) {
         this.fields = new Object[arity];
+        this.options = new HashMap<>();
```

Comment:
I prefer lazy create options map. https://github.com/apache/seatunnel/pull/7911/files#diff-80c522a8e538f221b1dc2c8f5fd2bed219470cc71ad6f43959923625983976f2R77 Because it useless at most time for now but it will created for each SeaTunnelRow.